<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>cedlGUIMain</class>
 <widget class="QMainWindow" name="cedlGUIMain">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>750</width>
    <height>825</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>cedlGUIMain</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="training_input_gb">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>311</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Training input parameters</string>
    </property>
    <widget class="QWidget" name="layoutWidget_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Input plate mapping file</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLineEdit" name="plate_mapping_train">
          <property name="text">
           <string>CecogExCedlDemo_plate_subset.txt</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pb_plate_mapping_train">
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Input CellH5 file</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLineEdit" name="ch5_input_train">
          <property name="text">
           <string>CecogExCedlDemo_plate_subset.ch5</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pb_ch5_input_train">
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Bounding box size (px)</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QSpinBox" name="bbox_size">
          <property name="minimum">
           <number>16</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="value">
           <number>40</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="training_gb">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>311</width>
      <height>561</height>
     </rect>
    </property>
    <property name="title">
     <string>Training parameters</string>
    </property>
    <widget class="QWidget" name="layoutWidget_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Trainer type</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QComboBox" name="learner">
          <item>
           <property name="text">
            <string>AdaGrad</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Nesterov</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Autoencoder layout (string)</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QLineEdit" name="autoencoder_architecture">
          <property name="text">
           <string>c16.5r_p2_c32.3r_p2_d256.1r_d64.0s</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="architecture_help">
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>?</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Learning rate</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QDoubleSpinBox" name="learning_rate">
          <property name="minimum">
           <double>0.010000000000000</double>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.005000000000000</double>
          </property>
          <property name="value">
           <double>0.050000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>180</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Momentum</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QDoubleSpinBox" name="momentum">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.005000000000000</double>
          </property>
          <property name="value">
           <double>0.900000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_10">
      <item>
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Minibatch size</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QSpinBox" name="batchsize">
          <property name="minimum">
           <number>4</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>128</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_11">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>290</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_11">
      <item>
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Input corruption level</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_12">
        <item>
         <widget class="QDoubleSpinBox" name="corruption">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.005000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_12">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>340</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_12">
      <item>
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Training epochs</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_13">
        <item>
         <widget class="QSpinBox" name="epochs">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>64</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_13">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>400</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_13">
      <item>
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Training samples</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_14">
        <item>
         <widget class="QSpinBox" name="nsamples">
          <property name="minimum">
           <number>1000</number>
          </property>
          <property name="maximum">
           <number>10000000</number>
          </property>
          <property name="singleStep">
           <number>10000</number>
          </property>
          <property name="value">
           <number>50000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_14">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>460</y>
       <width>281</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_14">
      <item>
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Group indicator to sample from (as in plate mapping file)</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_15">
        <item>
         <widget class="QLineEdit" name="neg_indicator">
          <property name="text">
           <string>neg</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="pb_start_training">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>520</y>
       <width>281</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Start feature training</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="log_window_gb">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>260</y>
      <width>401</width>
      <height>521</height>
     </rect>
    </property>
    <property name="title">
     <string>Output log</string>
    </property>
    <widget class="QPlainTextEdit" name="log_widget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>381</width>
       <height>481</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="encode_gb">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>10</y>
      <width>401</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Encoding parameters</string>
    </property>
    <widget class="QWidget" name="layoutWidget_15">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>371</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_15">
      <item>
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>Input plate mapping file</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_16">
        <item>
         <widget class="QLineEdit" name="plate_mapping_encode"/>
        </item>
        <item>
         <widget class="QPushButton" name="pb_plate_mapping_encode">
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_16">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>371</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_16">
      <item>
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Input CellH5 file</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_17">
        <item>
         <widget class="QLineEdit" name="ch5_input_encode"/>
        </item>
        <item>
         <widget class="QPushButton" name="pb_ch5_input_encode">
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="layoutWidget_17">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>371</width>
       <height>46</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_17">
      <item>
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>Autoencoder model directory (as generated in training)</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_18">
        <item>
         <widget class="QLineEdit" name="autoencoder_model"/>
        </item>
        <item>
         <widget class="QPushButton" name="pb_autoencoder_model">
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
    <widget class="QPushButton" name="pb_start_encode">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>371</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Start feature generation</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>750</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <action name="actionAdsf">
   <property name="text">
    <string>adsf</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
